cypher-shell/src/main/java/org/neo4j/shell/state/BoltStateHandler.java

```diff
@@ -0,0 +1,143 @@
+package org.neo4j.shell.state;
+
+import org.neo4j.driver.internal.logging.ConsoleLogging;
+import org.neo4j.driver.v1.*;
+import org.neo4j.shell.ConnectionConfig;
+import org.neo4j.shell.Connector;
+import org.neo4j.shell.TransactionHandler;
+import org.neo4j.shell.commands.Disconnect;
+import org.neo4j.shell.exception.CommandException;
+
+import javax.annotation.Nonnull;
+import java.util.logging.Level;
+
+/**
+ * Handles interactions with the driver
+ */
+public class BoltStateHandler implements TransactionHandler, Connector {
+    protected Driver driver;
+    protected Session session;
+    protected Transaction tx = null;
+
+    public BoltStateHandler() {
+    }
+
+    /**
+     * Returns an appropriate runner, depending on the current transaction state.
+     *
+     * @return a statementrunner to execute cypher
+     */
+    public StatementRunner getStatementRunner() {
+        if (tx != null) {
+            return tx;
+        }
+        return session;
+    }
+
+    @Override
+    public void beginTransaction() throws CommandException {
```

Comment:
We should considering letting *only* CypherShell to manipulate states in BoltStateHandler.